examples/remix/app/routes/buckets/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { ActionFunction, Form, useLoaderData } from "remix";
+import { s3Client } from "../../../backend/s3Client.server";
+import { createBucket, listBuckets } from "../../../backend/buckets.server";
+
+type BucketData = {
+    name: string;
+    createdAt: string;
+}
+
+type BucketsList = {
+    buckets: BucketData[];
+}
+
+export const action: ActionFunction = async ({ request }) => {
+    let { name } = Object.fromEntries(await request.formData());
+
+    if (typeof name !== "string") {
+        throw new Response("Invalid data submitted", { status: 400 });
+    }
+
+    await createBucket(s3Client, { Bucket: name as string });
+    console.log("created bucket", name);
+    return "success";
+};
+
+export async function loader() {
+    const buckets = await listBuckets(s3Client, {});
```

Comment:
We'll need to wrap this `await` in a try catch to properly display errors to the end user.